drivers/hmis/app/models/hmis/role.rb

```diff
@@ -44,54 +66,124 @@ def self.ensure_permissions_exist
     end
   end
 
+  def self.permissions_for_access(access)
+    permissions_with_descriptions.select { |_k, attrs| access == :viewable ? attrs[:access] == [:viewable] : attrs[:access].include?(access) }.keys
```

Comment:
why not?
```suggestion
    permissions_with_descriptions.select { |_k, attrs| attrs[:access].include?(access) }.keys
```

Reply:
You are correct, that's the right way to do it. However, this will make tests fail if a permission grants both `viewable` and `editable` access, as `can_administer_hmis` does. I'm going to adjust it so that `can_administer_hmis` only has `editable` access, which should fix the issue with the tests as that will make all accesses mutually exclusive. We can change this later once we're no longer relying on the `editable_by` and `viewable_by` scopes.

Reply:
Fixed